edx_lint/cmd/amnesty.py

```diff
@@ -0,0 +1,116 @@
+    """
+    for line in pylint_output:
+        if not line.strip() or line.startswith("****"):
+            continue
+
+        parsed = PYLINT_PARSEABLE_REGEX.search(line)
+        if parsed is None:
+            raise Exception(
+                "Unable to parse {!r}. Please run pylint with the "
+                "--output-format=parseable option".format(
+                    line
+                )
+            )
+
+        parsed_dict = parsed.groupdict()
+        parsed_dict['linenum'] = int(parsed_dict['linenum'])
+        yield PylintError(**parsed_dict)
+
+
+def format_pylint_disables(error_names, tag=True):
+    """
+    Format a list of error_names into a 'pylint: disable=' line.
+    """
+    tag_str = "  # lint-amnesty" if tag else ""
+    if error_names:
+        return "  # pylint: disable={}{}".format(
+            ", ".join(sorted(error_names)),
+            tag_str,
+        )
+    else:
+        return ""
+
+
+def fix_pylint(line, errors):
+    """
+    Yield any modified versions of ``line`` needed to address the errors in ``errors``.
+    """
+    if not errors:
+        yield line
+        return
+
+    current = PYLINT_EXCEPTION_REGEX.search(line)
+    if current:
+        original_errors = {disable.strip() for disable in current.group('disables').split(',')}
+    else:
+        original_errors = set()
+
+    disabled_errors = set(original_errors)
+
+    for error in errors:
+        if error.error_name == 'useless-suppression':
+            parsed = re.search("""Useless suppression of '(?P<error_name>[^']+)'""", error.error_msg)
+            disabled_errors.discard(parsed.group('error_name'))
+        elif error.error_name == 'missing-docstring' and error.error_msg == 'Missing module docstring':
+            yield format_pylint_disables({error.error_name}).strip()
+        else:
+            disabled_errors.add(error.error_name)
+
+    disable_string = format_pylint_disables(disabled_errors, not disabled_errors <= original_errors)
+
+    if current:
+        yield PYLINT_EXCEPTION_REGEX.sub(disable_string, line)
+    else:
+        yield re.sub(r'($\s*)', disable_string + r'\1', line)
+
+
+@click.command()
+@click.option(
+    '--pylint-output', default=sys.stdin, type=click.File(),
+    help="An input file containing pylint --output-format=parseable errors. Defaults to stdin."
+)
+def pylint_amnesty(pylint_output):
+    """
+    Add ``# pylint: disable`` clauses to add exceptions to all existing pylint errors in a codebase.
+    """
+    errors = defaultdict(lambda: defaultdict(set))
+    for pylint_error in parse_pylint_output(pylint_output):
+        errors[pylint_error.filename][pylint_error.linenum].add(pylint_error)
+
+    for file_with_errors in sorted(errors):
+        with open(file_with_errors) as input_file:
+            output_lines = []
+            for line_num, line in enumerate(input_file, start=1):
+                output_lines.extend(
+                    fix_pylint(
```

Comment:
Seems like it should take too much code to determine if it would be newly needed.  Otherwise, instructions will need to include running it twice?